DESIGN_GUIDELINES.md

```diff
@@ -1165,6 +1226,87 @@ use the SecretValue type [_awslint:secret-token_].
 * No need to put every class in a separate file. Try to think of a
   reader-friendly organization of your source files.
 
+### Levels of Abstraction
+
+There are three different levels of constructs in this library, beginning with
+low-level constructs, which we call _CFN Resources_ (or L1, short for "level 1")
+or Cfn (short for CloudFormation) resources. These constructs directly represent
+all resources available in AWS CloudFormation. CFN Resources are periodically
+generated from the AWS CloudFormation Resource Specification. They are named
+**Cfn**_Xyz_, where _Xyz_ is name of the resource. For example, CfnBucket
+represents the AWS::S3::Bucket AWS CloudFormation resource. When you use Cfn
+resources, you must explicitly configure all resource properties, which requires
+a complete understanding of the details of the underlying AWS CloudFormation
+resource model.
+
+The next level of constructs, L2, also represent AWS resources, but with a
+higher-level, intent-based API. They provide similar functionality, but provide
+the defaults, boilerplate, and glue logic you'd be writing yourself with a CFN
+Resource construct. AWS constructs offer convenient defaults and reduce the need
+to know all the details about the AWS resources they represent, while providing
+convenience methods that make it simpler to work with the resource. For example,
+the s3.Bucket class represents an Amazon S3 bucket with additional properties
+and methods, such as bucket.addLifeCycleRule(), which adds a lifecycle rule to
+the bucket.
+
```

Comment:
There's another level (sometimes referred to as 1.5) for "L2s with some sugar on top". They offer an API for a specific usage of an L2. Examples are "LambdaRestApi", "EdgeFunction", "rds.MySqlDatabase", "NodeJsFunction", "PythonFunction", etc.

I'd define L2.5 as a subclass of an L2 resource which offers a dedicated API for a specific common usage scenario of an L2. The RDS and Lambda examples are very strong.